validation_service_v2/validation_service/auth.py

```diff
@@ -94,13 +94,15 @@ async def get_collab_permissions_v2(collab_id, user_token):
         raise HTTPException(
             status_code=status.HTTP_401_UNAUTHORIZED, detail=userinfo["error_description"]
         )
-    target_team_name = f"collab-{collab_id}"
+    target_team_names = (f"collab-{collab_id}-{role}"
+                         for role in ("viewer", "editor", "administrator"))
     matching_teams = [
-        team for team in userinfo["roles"]["team"] if team.startswith(target_team_name)
+        team for team in userinfo["roles"]["team"] if team in target_team_names
     ]
     if len(matching_teams) == 0:
         permissions = {"VIEW": False, "UPDATE": False}
-    elif len(matching_teams) > 1:
+    elif len(matching_teams) > 1:  # this assumes a user only ever has one role,
```

Comment:
I am not sure we can make this assumption. I find that I have multiple roles for several Collabs such as:
'nmc-jupytertestcollab',
 'codejam-11-progamme-committee',
 'network-level-technical-coordination',
 'technical-coordination',
 'wp6-files-for-hbp'

My suggestions are:
(1) When we have multiple roles, there will be multiple matches for the same Collab. I don't think it is possible to have incorrect Collab matches with the updated matching implementation. So it should be safe to simply select any one of the matches and proceed.

(2) If for some reason it is not possible to uniquely identify a single Collab (if (1) does not hold true), I propose that we atleast keep a slightly different error message for 'len(matching_teams) > 1' case.... something like "Cannot uniquely identify the specified Collab!" . Currently this error message is the same as for the 'else...->  else' case.

Reply:
I prefer we go ahead with (1) as issue of multiple roles does crop up commonly for larger Collabs where certain users have specific roles individually, and also part of a group.